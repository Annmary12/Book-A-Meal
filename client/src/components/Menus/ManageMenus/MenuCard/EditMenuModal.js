import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import ReactModal from 'react-modal';
import MenuDetailsForm from '../MenuDetailsForm.js';
import styles from '../set-menu.scss';

const EditMenuModal = ({
  isOpen,
  handleClose,
  changeMenu,
  meals,
  date,
  categories
}) => (
  <ReactModal
    isOpen={isOpen}
    onRequestClose={handleClose}
    overlayClassName="react-modal-overlay"
    className={classnames('react-modal-body', styles.modal)}
    closeTimeoutMS={150}
  >
    <MenuDetailsForm
      closeModal={handleClose}
      handleSave={changeMenu}
      meals={meals}
      date={date}
      categories={categories}
    />
  </ReactModal>
);

EditMenuModal.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  handleClose: PropTypes.func.isRequired,
  changeMenu: PropTypes.func.isRequired,
  meals: PropTypes.arrayOf(PropTypes.object).isRequired,
  date: PropTypes.string.isRequired,
  categories: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default EditMenuModal;
