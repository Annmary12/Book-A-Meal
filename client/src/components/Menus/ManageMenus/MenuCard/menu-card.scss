@import '../../../../../assets/scss/_variables';
@import '../../../../../assets/scss/buttons';

.card {
  background-color: white;
  border-radius: 5px;
  margin: auto;
  display: flex;
  flex-direction: column;
  margin-bottom: 20px;
  cursor: pointer;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
  width: 300px;

  :global{
    animation: fadeInUp .2s ease-in;
  }

  &:hover{
    box-shadow: 0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);
  }
}

.dateHeader {
  text-align: center;
  margin: 0px;
}

.mealList {
  margin-top: 8px;
  font-size: 14px;
  span {
    margin-left: 10px;
  }
}

.categoryTitle {
  font-family: "Dancing script";
  margin: 0px;
  margin-bottom: 5px;
  font-size: 18px;
  text-transform: capitalize;
}

.footer{
  display: flex;
  padding: 10px;
  border-top: 1px solid #d0d0d0;
  max-height: 290px;
}
.body{
  padding: 10px;
  height: 179px;
  overflow: auto;
}

.actions {
  margin-left: auto;
}

.menuMeal {
  margin-bottom: 3px;
  font-size: 14px;
}

.category {
  margin-bottom: 10px;
}