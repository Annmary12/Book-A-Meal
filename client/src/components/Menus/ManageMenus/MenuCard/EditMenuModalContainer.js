import { connect } from 'react-redux';
import { changeMenu } from '../../../../actions/catererMenus.js';
import EditMenuModal from './EditMenuModal.js';

const mapStateToProps = state => ({
  meals: state.catererMeals.meals
});

export default connect(mapStateToProps, { changeMenu })(EditMenuModal);
