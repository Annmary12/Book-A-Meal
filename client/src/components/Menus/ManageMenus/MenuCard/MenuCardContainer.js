import { connect } from 'react-redux';
import { deleteMenu } from '../../../../actions/catererMenus.js';
import MenuCard from './MenuCard.js';

export default connect(null, { deleteMenu })(MenuCard);
