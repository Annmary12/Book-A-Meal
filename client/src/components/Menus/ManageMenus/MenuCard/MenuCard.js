import React, { Component } from 'react';
import PropTypes from 'prop-types';
import format from 'date-fns/format';
import ConfirmDeleteModal from '../../../ConfirmDeleteModal/ConfirmDeleteModal.js';
import cancelIcon from '../../../../../assets/img/cancel-icon.svg';
import pencilIcon from '../../../../../assets/img/pencil-icon.svg';
import styles from './menu-card.scss';
import EditMenuModalContainer from './EditMenuModalContainer.js';

/**
 * @class MenuCard
 */
class MenuCard extends Component {
  static propTypes = {
    menu: PropTypes.shape({
      date: PropTypes.string.isRequired,
      categories: PropTypes.array.isRequired
    }).isRequired,
    deleteMenu: PropTypes.func.isRequired
  }

  state = {
    deleteModalOpen: false,
    editModalOpen: false
  }

  openDeleteModal = () => {
    this.setState({ deleteModalOpen: true });
  }

  closeDeleteModal = () => {
    this.setState({ deleteModalOpen: false });
  }

  openEditModal = () => {
    this.setState({ editModalOpen: true });
  }

  closeEditModal = () => {
    this.setState({ editModalOpen: false });
  }

  /**
   * @returns {JSX} React JSX
   */
  render() {
    const { menu, deleteMenu } = this.props;
    return (
      <div className={styles.card}>
        <div className={styles.body}>
          <h4 className={styles.dateHeader}>
            {format(menu.date, 'Do MMM, YYYY')}.
          </h4>
          {
            menu.categories.map(category => (
              <div key={category.id} className={styles.category}>
                <h4 className={styles.categoryTitle}>
                - {category.title}
                </h4>
                {
                  category.meals.map(meal => (
                    <div
                      key={meal.id}
                      className={styles.menuMeal}
                    >
                      * {meal.title}
                    </div>
                  ))
                }
              </div>
            ))
          }
        </div>
        <div className={styles.footer}>
          <div className={styles.actions}>
            <button
              className={styles.iconBtn}
              onClick={this.openEditModal}
            >
              <img
                src={pencilIcon}
                width="18"
                alt="edit"
                className={styles.icon}
              />
            </button>
            <button
              className={styles.iconBtn}
              onClick={this.openDeleteModal}
            >
              <img
                src={cancelIcon}
                width="18"
                alt="delete"
                className={styles.icon}
              />
            </button>
          </div>
        </div>
        <ConfirmDeleteModal
          isOpen={this.state.deleteModalOpen}
          handleClose={this.closeDeleteModal}
          handleDelete={() => deleteMenu(menu.date)}
        />
        <EditMenuModalContainer
          isOpen={this.state.editModalOpen}
          handleClose={this.closeEditModal}
          date={menu.date}
          categories={menu.categories}
        />
      </div>
    );
  }
}

export default MenuCard;
