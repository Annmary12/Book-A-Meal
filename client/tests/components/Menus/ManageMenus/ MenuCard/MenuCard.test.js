import React from 'react';
import MenuCard from '../../../../../src/components/Menus/ManageMenus/MenuCard/MenuCard';

const props = {
  menu: {
    date: '2018-01-20',
    categories:[{
      title: 'test',
      meals: [{
        id: 'test-id',
        title: 'party rice',
        price: 300,
        img: 'image.png'
      }]
    }]
  },
  deleteMenu: jest.fn()
}
describe('MenuCard component', () => {
  const wrapper = shallow(
    <MenuCard
      {...props}
    />
  );

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('openDeleteModal method', () => {
    wrapper.instance().openDeleteModal();
    const { deleteModalOpen } = wrapper.instance().state;

    expect(deleteModalOpen).toBeTruthy();
  })
  describe('closeDeleteModal method', () => {
    wrapper.instance().closeDeleteModal();
    const { deleteModalOpen } = wrapper.instance().state;

    expect(deleteModalOpen).toBeFalsy();
  })
  describe('openEditModal method', () => {
    wrapper.instance().openEditModal();
    const { editModalOpen } = wrapper.instance().state;

    expect(editModalOpen).toBeTruthy();
  })
  describe('closeEditModal method', () => {
    wrapper.instance().closeEditModal();
    const { editModalOpen } = wrapper.instance().state;

    expect(editModalOpen).toBeFalsy();
  })
})