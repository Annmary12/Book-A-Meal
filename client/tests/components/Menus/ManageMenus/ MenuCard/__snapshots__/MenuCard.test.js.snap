// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuCard component should match snapshot 1`] = `
<div
  className="card"
>
  <div
    className="body"
  >
    <h4
      className="dateHeader"
    >
      20th Jan, 2018
      .
    </h4>
    <div
      className="category"
    >
      <h4
        className="categoryTitle"
      >
        - 
        test
      </h4>
      <div
        className="menuMeal"
        key="test-id"
      >
        * 
        party rice
      </div>
    </div>
  </div>
  <div
    className="footer"
  >
    <div
      className="actions"
    >
      <button
        className="iconBtn"
        onClick={[Function]}
      >
        <img
          alt="edit"
          className="icon"
          src="test-file-stub"
          width="18"
        />
      </button>
      <button
        className="iconBtn"
        onClick={[Function]}
      >
        <img
          alt="delete"
          className="icon"
          src="test-file-stub"
          width="18"
        />
      </button>
    </div>
  </div>
  <ConfirmDeleteModal
    confirmText="Are you sure you want to delete?"
    handleClose={[Function]}
    handleDelete={[Function]}
    isOpen={false}
  />
  <Connect(EditMenuModal)
    categories={
      Array [
        Object {
          "meals": Array [
            Object {
              "id": "test-id",
              "img": "image.png",
              "price": 300,
              "title": "party rice",
            },
          ],
          "title": "test",
        },
      ]
    }
    date="2018-01-20"
    handleClose={[Function]}
    isOpen={false}
  />
</div>
`;
